VSRAD.Package/Options/ProjectOptions.cs

```diff
@@ -72,63 +81,99 @@ public void UpdateActiveProfile(ProfileOptions newProfile)
 
-        public static ProjectOptions ReadLegacy(string path)
-        {
-            try
+            try // Try to load .profiles.json in the first place
+            {
+                var profiles = ProfileTransferManager.Import(profilesOptionsPath);
+                options.SetProfiles(profiles, options.ActiveProfile);
+            }
+            catch (FileNotFoundException)
             {
-                var legacyJson = JObject.Parse(File.ReadAllText(path));
-                return LegacyProfileImporter.ReadProjectOptions(legacyJson);
+                try // Try to load profiles from .conf.json if .profiles.json is missing
+                {
+                    var profiles = ProfileTransferManager.ImportObsolete(oldConfigPath);
+                    options.SetProfiles(profiles, options.ActiveProfile);
+                }
+                catch (FileNotFoundException) { } // Don't show an error if the configuration file is missing, just load defaults
+                catch (Exception e)
+                {
+                    Errors.ShowWarning($"An error has occurred while loading the profiles: {e.Message}\r\nProceeding with defaults.");
+                }
             }
             catch (Exception e)
             {
-                Errors.ShowWarning($"A legacy project options file was found but could not be converted: {e.Message}\r\nYou can transfer your configuration manually from {path}");
-                return new ProjectOptions();
+                Errors.ShowWarning($"An error has occurred while loading the profiles: {e.Message}\r\nProceeding with defaults.");
             }
+
+            if (options.Profiles.Count > 0 && !options.Profiles.ContainsKey(options.ActiveProfile))
+                options.ActiveProfile = options.Profiles.Keys.First();
+            return options;
         }
 
-        public void Write(string path)
+        public void Write(string visualConfigPath, string profilesConfigPath)
         {
             var serializedOptions = JsonConvert.SerializeObject(this, Formatting.Indented);
+            var serializedProfiles = JsonConvert.SerializeObject(Profiles, Formatting.Indented);
+
+            WriteAtomicWithErrorChecking(visualConfigPath, serializedOptions);
+            WriteAtomicWithErrorChecking(profilesConfigPath, serializedProfiles);
+        }
+
+        private static void WriteAtomicWithErrorChecking(string destPath, string contents)
+        {
             try
             {
-                WriteAtomic(path, serializedOptions);
+                WriteAtomic(destPath, contents);
             }
             catch (UnauthorizedAccessException)
             {
-                DialogResult res = MessageBox.Show($"RAD Debug is unable to save configuration, because {path} is read-only. Make it writable?", "Warning", MessageBoxButtons.OKCancel, MessageBoxIcon.Warning);
+                if (File.ReadAllText(destPath) == contents) return; // We don't want to show the warning if config haven't changed
```

Comment:
I think it's a good idea to wrap `File.ReadAllText` in a `try-catch` block as well. No need to display anything in case of an exception, just treat it as if the config is changed.

A small example of how to crash VS with this:
1. Deny `Read` and `Write` permissions on the config file for your user account
2. Open Visual Studio, navigate to the profile editor, click `Apply` without changing anything
3. Boom

Reply:
Thanks! Resolved in 0642b6b